examples/erosion2/storage.py

```diff
@@ -1,33 +1,64 @@
 import shelve
+import string
+import time
+
+
+_CHARS = sorted(string.ascii_lowercase + string.digits)
+_N_CHARS = len(_CHARS)
+
+
+def _encode_id(num):
+    alias = ""
+    while num:
+        alias = _CHARS[num % _N_CHARS] + alias
+        num //= len(_CHARS)
+    return alias if alias else _CHARS[0]
 
 
 class LinkDB:
     def __init__(self, db_path):
         self.db_path = db_path
-
-    def add_link(self, alias=None, *, target_url, expiry_time, max_count):
-        entry = {
-            "target": target_url,
-            "alias": alias,
-            "expiry_time": expiry_time,
-            "max_count": max_count,
-            "count": 0,
-        }
         with shelve.open(self.db_path) as db:
-            db[alias] = entry
+            db["last_id"] = 41660
+            db["entries"] = {}
```

Comment:
So is this resetting the database on restart?

Reply:
D'oh!